core/storage/suggestion/gae_models.py

```diff
@@ -821,7 +825,14 @@ def get_in_review_question_suggestions_by_offset(
                     break
                 for suggestion_model in suggestion_models:
                     offset += 1
-                    if suggestion_model.author_id != user_id:
+
+                    included_in_topics = False
+                    if filter_topic_skills is not None:
+                        if suggestion_model.target_id in filter_topic_skills:
+                            included_in_topics = True
+
+                    if suggestion_model.author_id != user_id and (
```

Comment:
Since you can check this first condition immediately, I would suggest just having the "target ID in filter topic skills" as a subpart of this if clause.

Also maybe make a frozenset of filter_topic_skills, then retrieval is a bit quicker than linear.

I.e.:
```
    if suggestion_model.author_id != user_id:
        if filter_topic_skills is not None and suggestion_model.target_id not in filter_topic_skills:
             continue
        ... do stuff here ...
```

Reply:
Sg!

Re the frozenset, I am not sure if it would increase retrieval performance, as per https://stackoverflow.com/questions/36555214/set-vs-frozenset-performance

Is Oppia implementation different?

Reply:
Resolving this thread for now, but feel free to reopen/create another regarding frozenset. Thanks!

Reply:
No strong pref between set and frozenset (I was intending more to compare set to list, the frozenset was just to keep it immutable). Thanks for checking!

Reply:
Thanks for the clarification! Makes sense. I just sent a commit to this PR with that improvement https://github.com/oppia/oppia/pull/17788/commits/ecd577cba18da99419c5a0794f4b3f23cbfc330b